odoo/modules/module.py

```diff
@@ -440,19 +444,34 @@ def get_modules_with_version():
             continue
     return res
 
-def adapt_version(version):
+
+def adapt_version(version: str) -> str:
+    """Return the canonical version string from a version in a manifest."""
+    version_parts = version.split('.')
+    if not (2 <= len(version_parts) <= 5):
+        raise ValueError(f"Invalid version {version!r}, must have beteen 2 and 5 parts")
```

Comment:
```suggestion
        raise ValueError(f"Invalid version {version!r}, must have between 2 and 5 parts")
```

Reply:
You can get a working registry today if you start with missing modules.
If the version is invalid, the result will be the same as if the module was not in the addons path. Whatever the change in the module, it will be ignored since the version is not compatible.